dask_cuda/explicit_comms/dataframe/shuffle.py

```diff
@@ -324,6 +471,15 @@ def shuffle(
     rank_to_inkeys = c.stage_keys(name=name, keys=df.__dask_keys__())
     c.client.cancel(df)
 
+    # Get batchsize
+    max_num_inkeys = max(len(k) for k in rank_to_inkeys.values())
+    batchsize = batchsize or dask.config.get("xcomm-batchsize", 1)
```

Comment:
If we're going to introduce some (or more?) configuration parameters for explicit comms, can we think about the naming hierarchy a bit? E.g. we now have `DASK_EXPLICIT_COMMS` to turn on use of explicit comms for the shuffle, but now `DASK_XCOMM_BATCHSIZE`. How are we to know these two are related?

Would prefer something hierarchical like:

```
dask.explicit-comms.enabled = True
dask.explicit-comms.batchsize = ...
```

And however that translates to environment variables.

Reply:
Good point, what about ?
```
dask.explicit-comms = True
dask.explicit-comms-batchsize = ...
```

Reply:
@wence- Perhaps we should consider how this choice would potentially interact with your proposal to make it possible to register the shuffle algorithm externally (e.g. https://github.com/dask/dask/pull/9521).  I don't have a clear design/pattern in my mind to suggest, but I thought it was worth mentioning.

Reply:
@rjzamora I think there are two slightly orthogonal things here:

1. How to enable the explicit comms shuffle
2. How to configure aspects of the explicit comms shuffle

Right now, we basically enable through dask config variables, and this PR introduces some configuration slot.

I agree that it would be nice to be able to register the shuffle externally, but I think that is slightly different.

Here, I'm just wondering about how we minimise the proliferation of different naming hierarchies in the dask configuration dictionary (and potentially make them more discoverable).

We could (not in this PR) collect all of the dask-cuda related configuration slots into a `dask-cuda.yaml` that we load and insert into the dask config (as `distributed` and `dask` itself do). This would also allow us to maintain the defaults in one place for all configuration options.

Reply:
> Good point, what about ?
> 
> ```
> dask.explicit-comms = True
> dask.explicit-comms-batchsize = ...
> ```

I think I am +epsilon in favour of hierarchical separation so `explicit-comms.batchsize`, `explicit-comms.enabled`, but I am not that fussed.

Reply:
Yeah, I totally agree that the two goals are (mostly) orthogonal. However, if we were to deprecate something like `dask.explicit-comms = True`, it seems reasonable to consider if the change would also make sense in the future when external registration is hopefully a thing. (I don't think it matters much, but wasn't sure)

Reply:
Agree, if https://github.com/dask/dask/pull/9521 gets in, it make sense to have common config hierarchy for all the shuffle backends.

But I think we should keep `dask.explicit-comms = True` for now.